server/core/runtime/models/shell_command_runner.go

```diff
@@ -33,10 +35,27 @@ type ShellCommandRunner struct {
 	outputHandler jobs.ProjectCommandOutputHandler
 	streamOutput  bool
 	cmd           *exec.Cmd
+	shell         *valid.CommandShell
 }
 
-func NewShellCommandRunner(command string, environ []string, workingDir string, streamOutput bool, outputHandler jobs.ProjectCommandOutputHandler) *ShellCommandRunner {
-	cmd := exec.Command("sh", "-c", command) // #nosec
+func NewShellCommandRunner(
+	shell *valid.CommandShell,
+	command string,
+	environ []string,
+	workingDir string,
+	streamOutput bool,
+	outputHandler jobs.ProjectCommandOutputHandler,
+) *ShellCommandRunner {
+	if shell == nil {
+		shell = &valid.CommandShell{
+			Shell:     "sh",
+			ShellArgs: []string{"-c"},
+		}
+	}
+	var args []string
+	args = append(args, shell.ShellArgs...)
+	args = append(args, command)
```

Comment:
It would be good to have a debug log entry here, detailing the shell and args. You will need to pass a logger object into this function.

Reply:
At this point the shell is getting set up and the execution is done in the `RunCommandAsync` method, where the project ctx logger is used to log the command. I updated the pre exec debug log and post exec info log now to reflect the change. Thanks for pointing this out. However, I would skip logging here on Init as this information is getting logged a bit later already. I added the log example to the PR description as you requested.

Reply:
Thanks @anryko, logs look good.